mobile/mobile-core/devices-service.ts

```diff
@@ -324,13 +363,59 @@ export class DevicesService implements Mobile.IDevicesService {
 				if (!this.$hostInfo.isDarwin && this._platform && this.$mobileHelper.isiOSPlatform(this._platform)) {
 					this.$errors.failWithoutHelp(message);
 				} else {
-					await this.startEmulator();
 					await this.executeCore(action, canExecute);
 				}
 			}
 		}
 	}
 
+	/**
+	 * Starts emulator or simulator if necessary depending on --device or --emulator flags.
+	 * If no options are passed runs default emulator/simulator if no devices are connected.
+	 * @param data mainly contains information about --emulator and --deviceId flags.
+	 */
+	public async startEmulatorIfNecessary(data?: Mobile.IDevicesServicesInitializationOptions): Promise<void> {
+		if (data && data.deviceId && data.emulator) {
+			this.$errors.failWithoutHelp(`--device and –emulator are incompatible options.
+If you are trying to run on specific emulator, use tns run –device <DeviceID>.`);
+		}
+
+		// are there any running devices
+		await this.detectCurrentlyAttachedDevices();
+		let foundDevices = this.getDeviceInstances();
+
+		//if no --device is passed and no devices are found, the default emulator is started
+		if (data && data.platform && !data.deviceId) {
```

Comment:
>why not fail with an error in the beginning if !data || !data.platform?

Because there's common code that will use this functionality and won't need platform specified